modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/events/LoginPostAction.java

```diff
@@ -286,4 +494,74 @@ private void _updateGuestCommerceOrder(
 	@Reference
 	private Portal _portal;
 
+	@Reference
+	private RoleLocalService _roleLocalService;
+
+	private class AccountInformationModel {
+
+		public AccountInformationModel(
+			int commerceSiteType, String cookieValue, User user) {
+
+			_populateModelFromCookie(cookieValue);
+
+			if (commerceSiteType == CommerceChannelConstants.SITE_TYPE_B2B) {
+				_accountType = AccountConstants.ACCOUNT_ENTRY_TYPE_BUSINESS;
+			}
+			else if (commerceSiteType ==
+						CommerceChannelConstants.SITE_TYPE_B2C) {
+
+				_accountType = AccountConstants.ACCOUNT_ENTRY_TYPE_PERSON;
+			}
+		}
+
+		public String getAccountName() {
+			return _accountName;
+		}
+
+		public String getAccountType() {
+			return _accountType;
+		}
+
+		public String getUserEmailAddress() {
+			return _userEmailAddress;
+		}
+
+		private String _extractValue(String keyValue) {
+			return StringUtil.extractLast(keyValue, StringPool.EQUAL);
+		}
+
+		private void _populateModelFromCookie(String cookieValue) {
+			String[] keyValues = cookieValue.split(StringPool.POUND);
+
+			for (String keyValue : keyValues) {
+				if (keyValue.startsWith("accountName=")) {
+					_accountName = _extractValue(keyValue);
+				}
+				else if (keyValue.startsWith("accountType=")) {
+					String accountType = _extractValue(keyValue);
+
+					if (accountType.equals(
+							AccountConstants.ACCOUNT_ENTRY_TYPE_BUSINESS) ||
+						accountType.equals(
+							AccountConstants.ACCOUNT_ENTRY_TYPE_PERSON)) {
+
+						_accountType = accountType;
+					}
+					else {
+						_accountType =
+							AccountConstants.ACCOUNT_ENTRY_TYPE_PERSON;
+					}
+				}
+				else if (keyValue.startsWith("userEmail=")) {
+					_userEmailAddress = keyValue;
+				}
+			}
```

Comment:
question: is the cookie content crypt in some way? Because I don't like to have a cookie with the email + accountName visible

Reply:
It's not, but it's bound to the domain with `sameSite` restriction.